dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -122,23 +122,37 @@ def __init__(
     def write(self) -> None:
         from dlt.common.libs.pyarrow import pyarrow as pa
         from dlt.common.libs.deltalake import (
+            DeltaTable,
             write_delta_table,
+            ensure_delta_compatible_arrow_table,
             _deltalake_storage_options,
             try_get_deltatable,
         )
 
+        # create Arrow table from Parquet files
+        # this eagerly materializes table in memory
+        # ideally, we'd use an Arrow dataset everywhere, but we need to
+        # explicitly check if there is data
+        # (https://github.com/delta-io/delta-rs/issues/2686)
         file_paths = [job.file_path for job in self.table_jobs]
-
-        if (
-            self.table["write_disposition"] == "merge"
-            and (
-                dt := try_get_deltatable(
-                    self.client.make_remote_uri(self.make_remote_path()),
-                    storage_options=_deltalake_storage_options(self.client.config),
+        arrow_table = pa.dataset.dataset(file_paths).to_table()
```

Comment:
on large loads this will probably fail or be super slow, no? there must be ways to cheat here, for example just load the first file and if there is something in there we know that we have rows and write the table the same way as before or something like that.